.github/actions/setup-uv-project/action.yml

```diff
@@ -0,0 +1,13 @@
+name: setup-uv-project
+description: Install uv and set up the project with its dependencies
+
+runs:
+  using: composite
+  steps:
+    - uses: astral-sh/setup-uv@v6
```

Comment:
`uv` setups live here, and I've activated the virtual environment by default, so we don't need to prefix commands with `uv`.